### PR TITLE
Use non-generic `TryComp()` for metadata & transform

### PR DESCRIPTION
**Pull request by ElectroJr**
Allows entity systems to use the cached xform & metadata queries

### CODE REVIEW

**Comment by Tayrtahn** [comment]
I didn't even realize this was a thing. Nice! 

Also, this looks like something that a code analyzer could catch people doing in the future. Maybe I'll work on that.

**Review by Emisse — APPROVED** [review]

File: Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs [hunk 1]
```diff
@@ -463,7 +463,7 @@ private void AfterInteract(EntityUid uid, StorageComponent storageComp, AfterInt
                 return;
             }
 
-            if (_xformQuery.TryGetComponent(uid, out var transformOwner) && TryComp<TransformComponent>(target, out var transformEnt))
+            if (TryComp(uid, out TransformComponent? transformOwner) && TryComp(target, out TransformComponent? transformEnt))
```
Comment (ShadowCommander) [inline Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs:466]: Not using xformQuery anymore?

**Review by ShadowCommander — COMMENTED** [review]
Looks good. Just one question.

File: Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs (hunk 1 above)
Comment (ElectroJr) [reply Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs:466]: The transform & metadata have non-generic variants that automatically use xform & metadata queries. i.e.,
```cs
protected bool TryComp(EntityUid uid, [NotNullWhen(true)] out TransformComponent? comp)
{
    return EntityManager.TransformQuery.TryGetComponent(uid, out comp);
}
```
The `Transform(EntityUid uid)` `Metadata(EntityUid uid) methods also use them, so there generally shouldn't be a need to use xform & metadata queries

**Pull request merged**